### DISCUSSION TITLE
Node.js Action cannot find config file with relative path

### DISCUSSION THREAD

**Discussion by Sheeri** [community/community#25454] [Actions]
<p>Hi! I’m adding the node.js Github Action and I’m having a very basic issue. This project has existed for a while, and the standard npm install/npm build (or even npm run) work as expected in a development environment.</p>
<p>In the Github Action,<br>
My builds fail due to:</p>
<pre><code class="lang-auto">Run npm run build --if-present
&gt; o-fish@0.1.0 build /home/runner/work/o-fish-web/o-fish-web
&gt; react-scripts build

Creating an optimized production build...

Failed to compile.

./src/services/stitch.service.js
Cannot find file '../config' in './src/services'.
</code></pre>
<p>The specific line in src/services/stitch.service.js is:<br>
import config from “…/config”;</p>
<p>The src/config.js file exists.</p>
<p>As this is an established project I’d rather not muck around with changing the import to be absolute instead of relative (e.g. something like ./src/config). I’d hate for others to have their builds start failing because I changed that.</p>
<p>I have read this page but it doesn’t address my issue:<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/guides/building-and-testing-nodejs" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="" height="">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/guides/building-and-testing-nodejs" target="_blank" rel="noopener nofollow ugc">Building and testing Node.js - GitHub Docs</a></h3>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>What am I missing?</p>
<p>If you need insight into the repository, it’s at<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/WildAid/o-fish-web/" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/1767528/181091187-c0da7db9-85a8-4ea7-a25c-62132e9969ec.png" class="thumbnail onebox-avatar" width="400" height="400">

<h3><a href="https://github.com/WildAid/o-fish-web/" target="_blank" rel="noopener nofollow ugc">WildAid/o-fish-web</a></h3>

<p>Web application for the Officer's Fishery Information Sharing Hub (O-FISH). The web app allows agencies to gain insights from the aggregated information gathered during a routine vessel inspect...</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
<br>
and the workflow is at: .github/workflows/node.js.yml (sorry, I can only post 2 links max because I’m new to the forums.)
<p>Thanks in advance for assistance!</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/sheeri">@Sheeri</a>,</p>
<p>The relative path in the script files should relative to the working directory of current step in the job of the workflow.<br>
By default, all the steps in a job are executing in the same default working directory (<a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="noopener nofollow ugc"><strong>github.workspace</strong></a>), and this also is the default location of your repository when using the <a href="https://github.com/actions/checkout" rel="noopener nofollow ugc"><strong>checkout</strong></a> action.</p>
<p>To explain this, here is an example:</p>
<ul>
<li>
<p>A <strong>.js</strong> file in the repository directory, its path is “<strong>./src/test-path.js</strong>”, see <a href="https://github.com/TestWorkflowsRML/JavaScript_lang/blob/073a7163b21b12fd3a0a6e11e6e5a902abf6cda6/src/test-path.js" rel="noopener nofollow ugc">here</a>.</p>
</li>
<li>
<p>A <strong>.sh</strong> file in the repository directory, its path is “<strong>./test.sh</strong>”, see <a href="https://github.com/TestWorkflowsRML/JavaScript_lang/blob/073a7163b21b12fd3a0a6e11e6e5a902abf6cda6/test.sh" rel="noopener nofollow ugc">here</a>. This bash shell script will be called in the <strong>.js</strong> file.</p>
</li>
<li>
<p>The workflow file in the repository directory, its path is “<strong>./.github/workflows/ci.yml</strong>”, see <a href="https://github.com/TestWorkflowsRML/JavaScript_lang/blob/073a7163b21b12fd3a0a6e11e6e5a902abf6cda6/.github/workflows/ci.yml" rel="noopener nofollow ugc">here</a>.</p>
</li>
</ul>
<pre><code class="lang-yaml">    steps:
      - name: Checkout
        uses: actions/checkout@v2.3.2

# Guess the working directory is the default "github.workspace" when execute the JavaScript file.
# Set the relative path of the bash shell script to be "./test.sh".
      - name: Execute src/test-path.js - 01
        run: node src/test-path.js
        env:
          SH_FILE_PATH: './test.sh'

# Guess the working directory is the directory of the JavaScript file (**./src**) when execute the JavaScript file.
# Set the relative path of the bash shell script to be "../test.sh".
      - name: Execute src/test-path.js - 02
        run: node src/test-path.js
        env:
          SH_FILE_PATH: '../test.sh'
</code></pre>
<p>The results:</p>
<ul>
<li>The step “<strong>Execute src/test-path.js - 01</strong>” can run successfully, see <a href="https://github.com/TestWorkflowsRML/JavaScript_lang/runs/1256814502?check_suite_focus=true#step:3:1" rel="noopener nofollow ugc">here</a>.</li>
<li>The step “<strong>Execute src/test-path.js - 02</strong>” run failed, see <a href="https://github.com/TestWorkflowsRML/JavaScript_lang/runs/1256814502?check_suite_focus=true#step:4:1" rel="noopener nofollow ugc">here</a>.</li>
</ul>

**Comment by Sheeri** [comment, marked as answer]
<p>This is a problem because when the repo is checked out, config.js doesn’t exist - config.js.tmpl does. So I had to add to the yaml file to copy the file and update it with the proper values.</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/sheeri">@Sheeri</a>,</p>
<blockquote>
<p>So I had to add to the yaml file to copy the file and update it with the proper values.</p>
</blockquote>
<p>No, you do not need to do this.<br>
You just need to use the <strong>working-directory</strong> keyword to re-set the working directory of current step, if you do not want to changed any in the <strong>.js</strong> file.<br>
For example:</p>
<pre><code class="lang-yaml">      - name: Execute src/test-path.js - 02
        working-directory: ./src
        run: node src/test-path.js
        env:
          SH_FILE_PATH: '../test.sh'
</code></pre>
<p>More details, you can see “<a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsrun" rel="noopener nofollow ugc"><strong>jobs.&lt;job_id&gt;.steps.run</strong></a>”.</p>

**Comment by Sheeri** [comment]
<p>We have a template file, so that people don’t accidentally check-in a config file.</p>
<p>e.g. config.js.tmpl is in the repo, config.js is not (it’s in .gitignore)</p>
<p>So I’m pretty sure I have to copy the config.js.tmpl first (or move it, to rename it).</p>
